data/blog/2022-11-07.retreat.md

```diff
@@ -0,0 +1,96 @@
+
+One afternoon we did a long [dragon dreaming](https://dragondreaming.org/) session to figure out the ideas and dreams for the future of MirageOS. Apart from bare-metal and MirageOS as hypervisor, we also dreamed about running your own digital infrastructure with MirageOS (with increased decentralization and security). Another afternoon we had a cocktail party at the pool.
+
+Some participants already wrote articles about their journey and activities, take a look at their writings: [Raphaël Proust](https://raphael-proust.gitlab.io/code/mirage-retreat-2022-10.html), [Jules, Sayo, Enguerrand, Sonja, Jan, Lucas](https://tarides.com/blog/2022-10-28-the-mirageos-retreat-a-journey-of-food-cats-and-unikernels), [Pierre](http://blog.enssat.fr/2022/10/pierre-alain-enssat-teacher-at-11th.html).
+
+#### OCaml 5 support
+
+OCaml 5 will soon be published, which has most of the runtime code being rewritten. The runtime also has many more dependencies. We worked on getting [ocaml-solo5](https://github.com/mirage/ocaml-solo5/) (the piece consisting of a math library, a C library, and the OCaml compiler for MirageOS unikernels) to use [OCaml 5](https://github.com/mirage/ocaml-solo5/pull/122). This is still work in progress, but already lead to [some](https://github.com/ocaml/ocaml/pull/11605) [upstream](https://github.com/ocaml/ocaml/pull/11606) [PRs](https://github.com/ocaml/ocaml/pull/11611) that are merged into the 5 branch as well.
+
+#### Integrating a Verified File System
+
+[FSCQ](https://github.com/mit-pdos/fscq) is a formally verified file system developed in the Coq proof assistant. So far MirageOS has a variety of read/write file systems: [chamelon](https://github.com/yomimono/chamelon/) (LittleFS) [tar](https://github.com/mirage/ocaml-tar/) (append-only), [fat](https://github.com/mirage/ocaml-fat/) (FAT16, barely used), and [one file FS](https://github.com/reynir/oneffs) (supporting only one file).
+
+At the retreat, we [first repaired some FSCQ proofs](https://github.com/mit-pdos/fscq/pull/17) to work with recent Coq releases, and then added [OCaml wrapper code and revised the OCaml extraction](https://github.com/mit-pdos/fscq/pull/18). The current state is that all the tests are passing, but we need to improve the performance of some operations. FSCQ was developed with lazyness in mind targetting Haskell, and uses lots of lists. In OCaml with strict evaluation the list operations are expensive.
```

Comment:
"The current state ..." ->

"The current state is that the most obvious bugs have been fixed, the filesystem is usable from FUSE and underwent basic testing, but performance needs to be improved. FSCQ was developed with lazyness in mind targeting Haskell; for instance, it uses lots of lists in a way that does not go well with OCaml's strict evaluation."

?

Reply:
thanks. I also linked to the issue 16 in their repository, which gives a pretty clear overview what is the current situation and may need some more work.